src/Filter/ChoiceFilter.php

```diff
@@ -35,29 +35,51 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
                 return;
             }
 
+            $isNullSelected = \in_array(null, $data['value'], true);
+            $data['value'] = array_filter($data['value'], static function ($value) {
+                return null !== $value;
+            });
+
             // Have to pass IN array value as parameter. See: http://www.doctrine-project.org/jira/browse/DDC-3759
             $completeField = sprintf('%s.%s', $alias, $field);
             $parameterName = $this->getNewParameterName($queryBuilder);
             if (ChoiceType::TYPE_NOT_CONTAINS === $data['type']) {
-                $this->applyWhere($queryBuilder, $queryBuilder->expr()->notIn($completeField, ':'.$parameterName));
+                $andConditions = [$queryBuilder->expr()->isNotNull($completeField)];
```

Comment:
This method is very long. Can you please try to extract some small methods. This improves reliability.

Reply:
Does 
```
if (\is_array($data['value'])) {
    $this->filterMultipleValues(...)
else {
    $this->filterSimpleValue()
}
```

would be enough ? @core23 @peter-gribanov